### PR TITLE
Layer progress

### PR DESCRIPTION
**Pull request by kennethjiang**

### CODE REVIEW

File: octoprint_obico/__init__.py [hunk 1]
```diff
@@ -540,6 +598,15 @@ def is_blueprint_protected(self):
     def is_pro_user(self):
         return self.linked_printer.get('is_pro')
 
+    def createFilePreProcessor(self, path, file_object, blinks=None, printer_profile=None, allow_overwrite=True, *args, **kwargs):
```
Comment (kennethjiang) [inline octoprint_obico/__init__.py:598]: @aarontuftss When we copied from places, we should do a reasonable amount of clean up to keep the maintainability of our app. Things like naming convention (camel-case vs snake-case) is a low-hanging fruits for such cleanup.

File: octoprint_obico/preprocessor.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import re
+import sys
+import octoprint.filemanager
+
+
+
+__python_version__ = 3 if sys.version_info >= (3, 0) else 2
+
+
```
Comment (kennethjiang) [inline octoprint_obico/preprocessor.py:9]: @aarontuftss Why so many empty lines?

Comment (aarontuftss) [reply octoprint_obico/preprocessor.py:9]: Spacing. Felt cleaner but I see how it is unnecessary.

File: octoprint_obico/__init__.py (hunk 1 above)
Comment (aarontuftss) [reply octoprint_obico/__init__.py:598]: Ahh noted. Seems I have been writing too much JS & forgot I was in python w/ the naming conventions.

**Pull request merged**